cpp/src/io/json/reader_impl.cu

```diff
@@ -592,12 +594,22 @@ table_with_metadata reader::impl::convert_data_to_table(cudaStream_t stream)
   CUDA_TRY(cudaGetLastError());
 
   // postprocess columns
+  auto target = make_strings_column(std::vector<char>{'\\', '\"', '\\', '\\'}, {0, 2, 4});
+  auto repl   = make_strings_column({'\"', '\\'}, {0, 1, 2});
+
   thrust::host_vector<cudf::size_type> h_valid_counts = d_valid_counts;
   std::vector<std::unique_ptr<column>> out_columns;
   for (size_t i = 0; i < num_columns; ++i) {
     out_buffers[i].null_count() = num_records - h_valid_counts[i];
 
-    out_columns.emplace_back(make_column(out_buffers[i]));
+    auto out_column = make_column(out_buffers[i], stream, mr_);
+    if (out_column->type().id() == type_id::STRING) {
+      // Need to remove escape character in case of '\"' and '\\'
+      out_columns.emplace_back(cudf::strings::detail::replace(
+        out_column->view(), target->view(), repl->view(), mr_, stream));
```

Comment:
I know this may be out of scope for this PR but essentially there are 3 copies of the strings for this column in device memory at this point. The first is the original input strings scattered throughout device memory in JSON format. The second is those strings joined together into a strings column created on line `auto out_column = make_column(out_buffers[i], stream, mr_);` above. The third is the one created here with the filtered out escape sequences. The second one is freed right away at the bottom of the for-loop but for a brief moment ... we've reduced the size of an input JSON we can now process on the GPU. 

Perhaps there is away to remove the escaped sequences when the data is copied into contiguous chars column in the first step and avoid the third copy. I guess it bothers me because the number of escaped characters is probably very small compared to the total number of characters and may not even appear at all in the input but the column is copied regardless.

I know we do this in the CSV parser (and similarly in the writers) as well for escape sequences there so perhaps this concern should be created as a github issue to be fixed for both.
https://github.com/rapidsai/cudf/blob/cb6fb6fda58f52afb346e71a692813bffc255148/cpp/src/io/csv/reader_impl.cu#L375-L376

Reply:
Thank you, I wanted to delete the old copy once I created new one, but forgot to do so.

And I agree that these needs to be taken care while copying data, but as several other readers and writers are depedent on this, I hesitated to make any changes over there. I will create a github issue to address this issue.

Reply:
Created a feature request for this https://github.com/rapidsai/cudf/issues/6305